presto-main/src/main/java/com/facebook/presto/connector/informationSchema/InformationSchemaMetadata.java

```diff
@@ -190,11 +207,56 @@ public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTable
     @Override
     public List<ConnectorTableLayoutResult> getTableLayouts(ConnectorSession session, ConnectorTableHandle table, Constraint<ColumnHandle> constraint, Optional<Set<ColumnHandle>> desiredColumns)
     {
-        InformationSchemaTableHandle handle = (InformationSchemaTableHandle) table;
-        ConnectorTableLayout layout = new ConnectorTableLayout(new InformationSchemaTableLayoutHandle(handle, constraint.getSummary()));
+        if (constraint.getSummary().isNone()) {
+            return ImmutableList.of();
+        }
+
+        InformationSchemaTableHandle handle = checkTableHandle(table);
+
+        Set<QualifiedTablePrefix> prefixes = calculatePrefixesWithSchemaName(session, constraint.getSummary(), constraint.predicate());
+        if (!handle.getSchemaTableName().equals(TABLE_SCHEMATA)) {
```

Comment:
Do you assume that every information schema table other than `schemata` is based on enumeration of tables and contains `table_catalog`, `table_schema` and `table_name` columns? 
Maybe extract this check to a `isTablesEnumeratingTable` method, where you can return false for "schemata", true for 3 other tables and failure otherwise?